components/reddit/reddit.app.js

```diff
@@ -0,0 +1,191 @@
+          });
+        }
+        return options;
+      },
+    },
+    username: {
+      type: "string",
+      label: "Username",
+      description: "The username you'd like to watch.",
+    },
+    t: {
+      type: "string",
+      label: "t",
+      description: "one of (hour, day, week, month, year, all)",
+      options: ["hour", "day", "week", "month", "year", "all"],
+    },
+    sr_detail: {
+      type: "boolean",
+      label: "Include Subreddit details?",
+      description:
+        "If set to true, includes details on the subreddit in the emitted event.",
+      default: false,
+      optional: true,
+    },
+  },
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    async getNewHotSubredditPosts(subreddit, g, show, sr_detail, limit = 100) {
+      const params = {};
+      if (show) {
+        params["show"] = "all";
+      }
+      params["g"] = g;
+      params["sr_detail"] = sr_detail;
+      params["limit"] = limit;
+      return await this._makeRequest({
+        path: `/r/${subreddit}/hot`,
+        params,
+      });
+    },
+    async getNewSubredditLinks(before_link, subreddit, limit = 100) {
+      return await this._makeRequest({
+        path: `/r/${subreddit}/new`,
+        params: {
+          before: before_link,
+          limit,
+        },
+      });
+    },
+    async getNewSubredditComments(
+      subreddit,
+      subreddit_id36,
+      context,
+      depth,
+      sr_detail,
+      limit = 100
+    ) {
+      const resp = await this._makeRequest({
+        path: `/r/${subreddit}/comments/article`,
+        params: {
+          article: subreddit_id36,
+          context,
+          depth,
+          limit,
+          sort: "new",
+          sr_detail,
+          theme: "default",
+          threaded: true,
+          trucate: 0,
+        },
+      });
+      return resp[1];
+    },
+    async getNewUserLinks(
+      before,
+      username,
+      context,
+      t,
+      sr_detail,
+      limit = 100
+    ) {
+      const params = {
+        before,
+        context,
+        show: "given",
+        sort: "new",
+        t,
+        type: "links",
+        sr_detail,
+        limit,
+      };
+
+      return await this._makeRequest({
+        path: `/user/${username}/submitted`,
+        params,
+      });
+    },
+    async getNewUserComments(
+      before,
+      username,
+      context,
+      t,
+      sr_detail,
+      limit = 100
+    ) {
+      const params = {
+        before,
```

Comment:
Shouldn't we use `after` if we're interested in the new comments?

Reply:
No, it is because of how new comments are returned by Reddit API - ordered first, so we technically iterate from new records to older ones. That's why we use `before` instead. This was discussed as well with Dylan previously.